### PR TITLE
snake_case client api

### PR DESCRIPTION
**Pull request by dcolligan**
Issue #1260

### CODE REVIEW

File: ga4gh/cli.py [hunk 1]
```diff
@@ -338,7 +340,7 @@ def __init__(self, args):
         self._md5checksum = args.md5checksum
 
     def _run(self, referenceSetId):
-        iterator = self._client.searchReferences(
+        iterator = self._client.search_references(
             accession=self._accession, md5checksum=self._md5checksum,
             referenceSetId=referenceSetId)
```
Comment (jeromekelleher) [inline ga4gh/cli.py]: We need to snake_case the function parameters too: ``reference_set_id`` here.

File: ga4gh/client.py [hunk 2]
```diff
@@ -543,10 +549,11 @@ def searchIndividuals(self, datasetId, name=None):
         request.dataset_id = datasetId
         request.name = pb.string(name)
         request.page_size = pb.int(self._pageSize)
-        return self._runSearchRequest(
+        return self._run_search_request(
             request, "individuals", protocol.SearchIndividualsResponse)
 
-    def searchReadGroupSets(self, datasetId, name=None, bioSampleId=None):
+    def search_read_group_sets(
+            self, datasetId, name=None, bioSampleId=None):
```
Comment (jeromekelleher) [inline ga4gh/client.py]: So, for keyword parameters like these we do need to snake_case to ``bio_sample_id`` or it'll be very confusing. I think it's as well to just snake_case this entire module and be done with it.

**Comment by jeromekelleher** [comment]
Fantastic @dcolligan, thanks for getting stuck into this! I think we will need to snake_case the full client module, or it'll be a real mess for clients to use. Otherwise, looking great.

File: docs/demo.rst [hunk 3]
```diff
@@ -189,13 +189,13 @@ performed above, we can use the following code:
...
+    datasets = list(httpClient.search_datasets())
     # Get the variantSets in the first dataset.
-    variantSets = list(httpClient.searchVariantSets(
+    variantSets = list(httpClient.search_variant_sets(
         datasetId=datasets[0].id))
     # Now get the variants in the interval [45000, 50000) on chromosome 1
     # in the first variantSet.
-    iterator = httpClient.searchVariants(
+    iterator = httpClient.search_variants(
         variantSetId=variantSets[0].id,
```
Comment (jeromekelleher) [inline docs/demo.rst]: snake_case variantSetId etc.

**Comment by jeromekelleher** [comment]
Great work, thanks @dcolligan. +1 modulo minor issue above.

**Comment by dcolligan** [comment]
fixed and merging

**Pull request merged**